### ISSUE TITLE
Time to mine target amount

### ISSUE THREAD

**Issue by JoshSchreuder** [DenysAshikhin/gameplay-planner#42]
Would be nice to have a way to enter a target amount of a mineral, and show how much longer it will take at the current mining rates, to know when building will become available

Eg. I need 320 gold. I have 285 banked in my save, and it will take X hours to mine the remaining based on the current miner allocation.

It is possible to work this out by using the mining rates but having a UI for this would be handy IMO

**Comment by DenysAshikhin** [comment]
I see your point, and definetily feel you. However, I can't think of a good way to do this in a simple UI without making it a bit complicated? As it is, if you know it will take x hours to reach y% you can do napkin math to get the time to get certain resources. I know it's not what you want and you are asking for a UI, but I feel it would clutter it a bit.

If you have a mockup you have in mind, share it here and we can discuss. If we can find something we agree on I'll make an actual UI for it!

**Comment by JoshSchreuder** [comment]
Here's what I was imagining, not sure if it is possible with this small amount of user input and what's available in the save file
![image](https://github.com/user-attachments/assets/b7edf824-8c71-44ac-9b26-464f5fb61332)

Thinking about it more, it may not be possible to accurately determine the time given the decrease in materials / miner bonuses. But it might still be useful as a rough estimate (I put an (i) icon here to mention this)

**Comment by DenysAshikhin** [comment]
![image](https://github.com/user-attachments/assets/cf700408-b27d-41e2-a15e-2d3430a2f877)
So I have attached a 1080p screenshot. Another standard I do my best to absolutely abide by, is making sure everything fits on a 1080p (you're 100% using a 2k :p ) screen with no scrolling (not always possible but I try).

There's not really enough space to the right. However, I *could* place it below the list and it would fit...mostly unless you have > 8 outposts.
Would the time to target be based on your current assignment of miners?

**Comment by JoshSchreuder** [comment]
> Would the time to target be based on your current assignment of miners?

Yeah I think that would be fine - working out the placement of the miners to get the material you want is kind of the easy part, just being able to see a rough time to mine would be useful I think.

Having it fit a 1080p screen with no scrolling is a lofty goal, I respect it 😄

**Comment by DenysAshikhin** [comment]
Okay, I don't think I'll get to it this weekend, but in a week or two I should hopefully finish this!
